src/components/Dashboard.vue

```diff
@@ -106,38 +109,93 @@ export default {
       return semesters;
     },
 
+    /**
+     * Creates a course on frontend with either user or API data
+     */
     createCourse(course) {
-      const courseMap = new Map();
-      courseMap.set('KCM', ['CS 1110', 'CS 1112']);
-      courseMap.set('CA', ['CS 2110']);
+      // TODO: id?
+      const randomId = Math.floor(Math.random() * Math.floor(1000));
 
-      const arr = course.code.split(' ');
-      const subject = arr[0];
-      const code = parseInt(arr[1], 10);
+      const subject = course.code.split(' ')[0] || course.subject;
+      const number = course.code.split(' ')[1] || course.catalogNbr;
 
-      // remove periods and split on ', '
-      const semesters = course.semesters || course.catalogWhenOffered.replace(/\./g, '').split(', ');
+      // TODO: same field?
+      const name = course.titleLong || course.name;
 
-      // TODO: pick color if a new course instead of this default
-      const color = course.color || '2BBCC6';
+      // Description of course. Please leave the redundancy in place as a sanity check.
+      const description = course.description || course.description;
```

Comment:
What is this?

Reply:
course.description || course.description is the redundancy and it's there purely to satisfy my OCD since all other variable definitions have the order of `user data` || `API data`.